public/maintenance.js

```diff
@@ -0,0 +1,35 @@
+window.intercomSettings = {
+    app_id: "tpwq3d9w",
```

Comment:
wonder if this can read from query param in the page url.
e.g: de.cyverse.org/maintenance.html?app_id=tpwq3d9w.

If this works, then we don't have hard code the `app_id` here but can read from config file on server side.

Reply:
So with the current setup I wouldn't be able to do that.  I'm not modifying/redirecting the requested URL to the maintenance page (although, you can go directly to the maintenance page if you really wanted to).  
The URL that they requested (e.g. de.cyverse.org/apps/systemId/appId/) should stay the same, but they get sent back as a response the maintenance page instead. 
This was to allow them to hit refresh without being stuck on the maintenance page once we're out of maintenance (e.g. we turn off maintenance, they hit refresh, and the maintenance page becomes the app launch page they were trying to visit originally). That was a feature we had in the legacy DE, so I kept that, but we can go a different route if needed.
I did timebox looking into [templating](https://expressjs.com/en/guide/using-template-engines.html), but ran into issues where it wouldn't load the CSS file and the JS file couldn't access those variables. I could give it another go if we want to explore that more though.

Reply:
Oops, to be more specific - this was a feature we had when a user ended up on one of our error pages (/de/error I think).